src/Symfony/Component/Serializer/Context/Encoder/CsvEncoderContextBuilder.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Context\Encoder;
+
+use Symfony\Component\Serializer\Context\ContextBuilder;
+use Symfony\Component\Serializer\Encoder\CsvEncoder;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * A helper providing autocompletion for available CsvEncoder options.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+class CsvEncoderContextBuilder extends ContextBuilder
+{
+    /**
+     * Configures the column delimiter character.
+     *
+     * Must be one character only.
+     *
+     * @throws InvalidArgumentException
+     */
+    public function withDelimiter(?string $delimiter): static
+    {
+        if (null !== $delimiter && \strlen($delimiter) > 1) {
+            throw new InvalidArgumentException(sprintf('The "%s" delimiter must be one character only.', $delimiter));
+        }
+
+        return $this->with(CsvEncoder::DELIMITER_KEY, $delimiter);
+    }
+
+    /**
+     * Configures the field enclosure character.
+     *
+     * Must be one character only.
+     *
+     * @throws InvalidArgumentException
+     */
+    public function withEnclosure(?string $enclosure): static
+    {
+        if (null !== $enclosure && \strlen($enclosure) > 1) {
```

Comment:
Should we allow an empty enclosure? Does it work? Same for the separator.

Reply:
According to `fputcsv`, delimiter and enclosure must be a single char whereas escape char must be empty or a single char.